includes/utils/class-amp-validation-utils.php

```diff
@@ -535,64 +532,44 @@ public static function print_edit_form_validation_status( $post ) {
 			return;
 		}
 
-		$url                    = null;
-		$validation_status_post = null;
-		$validation_errors      = array();
+		// Skip if the post type is not viewable on the frontend, since we need a permalink to validate.
+		if ( ! is_post_type_viewable( $post->post_type ) ) {
+			return;
+		}
 
-		// Incorporate frontend validation status if there is a known URL for the post.
-		if ( is_post_type_viewable( $post->post_type ) ) {
-			$url = amp_get_permalink( $post->ID );
+		$url                    = amp_get_permalink( $post->ID );
+		$validation_errors      = array();
+		$validation_status_post = self::get_validation_status_post( $url );
 
-			$validation_status_post = self::get_validation_status_post( $url );
-			if ( $validation_status_post ) {
-				$data = json_decode( $validation_status_post->post_content, true );
-				if ( is_array( $data ) ) {
-					$validation_errors = array_merge( $validation_errors, $data );
-				}
-			}
+		// No validation status exists yet, so there is nothing to show.
+		if ( ! $validation_status_post ) {
+			return;
 		}
 
-		// If no results from URL are available, validate post content outside frontend context.
-		if ( empty( $validation_errors ) && post_type_supports( $post->post_type, 'editor' ) ) {
-			self::process_markup( $post->post_content );
-			$validation_errors = array_merge(
-				$validation_errors,
-				self::$validation_errors
-			);
-			self::reset_validation_results();
-
-			// Make sure original post is restored after applying shortcodes which could change it.
-			$GLOBALS['post'] = $post; // WPCS: override ok.
-			setup_postdata( $post );
+		if ( $validation_status_post ) {
```

Comment:
Could this be an else block of the earlier `if block?
```php
if ( ! $validation_status_post ) {
	return;
} else { 
	$validation_errors = json_decode( $validation_status_post->post_content, true );
}
```

Reply:
Actually, there's no need for the `if` at all here. Good point.